### PR TITLE
[Improvement][Starter] Use the public CommandLine util class to parse the args 

### PR DESCRIPTION
**Pull request by zhuangchong**
## Purpose of this pull request

Use the public CommandLine util class to parse the parameters without creating a separate one under the Flink and Spark startup modules.

change:

```
create seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/core/base/utils/CommandLineUtils.java
delete  seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/utils/CommandLineUtils.java
delete  seatunnel-core/seatunnel-core-spark/src/main/java/org/apache/seatunnel/core/spark/utils/CommandLineUtils.java

create  seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/utils/CommandLineUtils.java
delete  seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/utils/CommandLineUtils.java
delete  seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/utils/CommandLineUtils.java
```

CommandLineUtils parse args exception, print:

```
deploy-mode: cluster2xxx is not allowed.
Usage: <main class> [options]
  Options:
    -t, --check       check config (default: false)
  * -c, --config      Config file
  * -e, --deploy-mode Spark deploy mode (values: [CLIENT, CLUSTER])
    -h, --help        Show the usage message
  * -m, --master      Spark master
    -i, --variable    variable substitution, such as -i city=beijing, or -i 
                      date=20190318 (default: [])
```

parse error

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/core/base/utils/CommandLineUtils.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        throw new UnsupportedOperationException("CommandLineUtils is a utility class and cannot be instantiated");
+    }
+
+    public static <T extends AbstractCommandArgs> T parse(String[] args, T obj) {
+        return parse(args, obj, null, false);
+    }
+
+    public static <T extends AbstractCommandArgs> T parse(String[] args, T obj, String programName, boolean acceptUnknownOptions) {
+        JCommander jCommander = JCommander.newBuilder()
+                .programName(programName)
```
Comment (hailin0) [inline seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/core/base/utils/CommandLineUtils.java:40]: Ignore `programName` null value ?

**Review by hailin0 — COMMENTED** [review]
LGTM

File: seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/core/base/utils/CommandLineUtils.java (hunk 1 above)
Comment (zhuangchong) [reply seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/core/base/utils/CommandLineUtils.java:40]: null values ​​have no effect.

File: seatunnel-core/seatunnel-core-flink/src/test/java/org/apache/seatunnel/core/flink/utils/CommandLineUtilsTest.java [hunk 2]
```diff
@@ -17,67 +17,33 @@
...
             "-r", "run-application", "--unkown", "unkown-command"};
-        flinkCommandArgs = CommandLineUtils.parseCommandArgs(args1, FlinkJobType.SQL);
-        Assertions.assertEquals(flinkCommandArgs.getFlinkParams(), Arrays.asList("--detached", "--unkown", "unkown-command"));
+        flinkCommandArgs = CommandLineUtils.parse(args1, new FlinkCommandArgs(), FlinkJobType.SQL.getType(), true);
+        Assertions.assertEquals(flinkCommandArgs.getOriginalParameters(), Arrays.asList("--detached", "--unkown", "unkown-command"));
         Assertions.assertEquals(flinkCommandArgs.getRunMode(), FlinkRunMode.APPLICATION_RUN);
         Assertions.assertEquals(flinkCommandArgs.getVariables(), Arrays.asList("city=shenyang", "date=20200202"));
     }
 
-    @Test
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-core-flink/src/test/java/org/apache/seatunnel/core/flink/utils/CommandLineUtilsTest.java:52]: Why remove this test case？

Comment (zhuangchong) [reply seatunnel-core/seatunnel-core-flink/src/test/java/org/apache/seatunnel/core/flink/utils/CommandLineUtilsTest.java:52]: The CommandLineUtils.buildFlinkCommand method has been moved to the FlinkStarter buildCommands method. The buildCommands method has a corresponding test class, so this part of the test method has been deleted.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Pull request merged**